### ISSUE TITLE
(ecs-patterns): Expose arrays of load balancers, listeners, and targetGroups as properties in ApplicationMultipleTargetGroupsFargateService and other patterns constructs

### ISSUE THREAD

**Issue by evanj354** [aws/aws-cdk#14735]
I want to utilize the construct *ApplicationMultipleTargetGroupsFargateService* for creating a Fargate service fronted by an Application Load Balancer. I need the Application Load Balancer to have 2 listeners and each listener to forward traffic to a distinct ApplicationTargetGroup. 

The problem I run is after I instantiate the construct, and want to configure health checks for each individual target group. The *ApplicationMultipleTargetGroupsFargateService* provides a *targetGroup* property, however, according to the CDK documentation, this only contains a reference to the default Target Group. Ideally, this would contain a list of references to the different ApplicationTargetGroups configured within the construct. Is there any method for getting a reference to these additional target groups in the CDK from the construct for configuring health checks? 

```
const service = new ApplicationMultipleTargetGroupsFargateService(this, 'ALBFargateService', {
    cluster: this.ecsALBCluster,
    taskDefinition: taskDefinition,
    desiredCount,
    assignPublicIp: true, // needed for outbound network connectivity in a public subnet
    serviceName: 'ALBFargateService',
    loadBalancers: [
        {
            name: 'ApplicationLoadBalancer',
            listeners: [
                {
                    name: 'TrafficListener',
                    port: 443,
                    protocol: ApplicationProtocol.HTTP
                },
                {
                    name: 'HealthCheckListener',
                    port: 8050,
                    protocol: ApplicationProtocol.HTTP
                }
            ]
        }
    ],
    targetGroups: [
        {
            listener: 'TrafficListener',
            containerPort: 443,
            protocol: Protocol.TCP
        },
        {
            listener: 'HealthCheckListener',
            containerPort: 8050,
            protocol: Protocol.TCP
        }
    ]
});

service.targetGroup.configureHealthCheck({
    port: '8050',
    protocol: elbv2.Protocol.HTTP,
    healthyThresholdCount: 2,
    unhealthyThresholdCount: 2,
    timeout: Duration.seconds(10),
    interval: Duration.seconds(30),
    healthyHttpCodes: '200'
});

// TODO: Configure Health Checks for second target group on port 8050
```

As shown above, I am configuring the service with 2 listeners and 2 target groups associated with a respective listener. Only the first target group's reference is accessible via the targetGroup property.

**Comment by peterwoodworth** [comment]
Hey @evanj354, sorry for the long response time.

What happens when you create multiple target groups is this function is called with the props you passed in as the `targetProps`

https://github.com/aws/aws-cdk/blob/9d01b4fabdf50a1e6691c054a674d768e5816a3c/packages/%40aws-cdk/aws-ecs-patterns/lib/base/application-multiple-target-groups-service-base.ts#L462-L483

On line 464 you can see that the CDK does add the targets to their respective listeners. However, the return statement only returns the first targetGroup you submitted. You can only view one Listener at a time as well, making it so that you can directly expose these properties on the construct is completely reasonable. I'm relabelling this as a feature request

**Comment by rohitingle9** [comment]
@peterwoodworth,

How can we configure the health checks (or other properties) for the non-default target group today?

**Comment by peterwoodworth** [comment]
I don't think you're currently able to do configure non-default target groups with this construct unfortunately. I might be wrong, but I wasn't able to find any ways to do this

**Comment by peterwoodworth** [comment]
You can work around this by using [escape hatches](https://docs.aws.amazon.com/cdk/latest/guide/cfn_layer.html)

```
const lb = service.node.children[x] as ApplicationLoadBalancer
console.log(lb.listeners)
```
`x` is whatever number child your ApplicationLoadBalancer is. You can see this by doing `console.log(service.node.children)`

**Comment by peterwoodworth** [comment]
Another customer messaged me about this issue and not being able to access their second load balancer on `NetworkMultipleTargetGroupsFargateService` so I'm updating the issue. Workarounds to not finding any other children are possible through escape hatches like the example above

**Comment by evanj354** [comment]
In this case, the above method of using escape hatches won't work for exposing the listeners. According to [CDK documentation](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-elasticloadbalancingv2.ApplicationLoadBalancer.html#listeners), the property listeners on ApplicationLoadBalancer construct is only valid for owned construct. I suspect this means that it is only valid for constructs that are explicitly defined, not implicitly created as a dependency of another construct. I tested this and confirmed that the load balancer does indeed not have a property for listeners. 

I was however, able to use the following escape hatch workaround to expose the listeners:

```javascript
const applicationListeners = service.node.findChild('ApplicationLoadBalancer').node.children.filter(child  => { return (child as CfnResource).toString().includes('Listener') });
applicationListeners.forEach((listener: IConstruct) => {
    const cfnListener = listener.node.defaultChild as elbv2.CfnListener;
});
```
Obviously, this logic is convoluted, and I was unable to use escape hatches to access to other components of the construct (i.e. target groups). One thing I did notice, is that the parent of this construct [ApplicationMultipleTargetGroupsServiceBase](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ecs-patterns.ApplicationMultipleTargetGroupsServiceBase.html) does have properties for ```listeners``` and ```targetGroups```. Unfortunately, these properties are protected, so they can not be reference outside of the child construct. I think the simple fix here would be to set these properties to public readonly, so they can be accessed via the child construct. This would eliminate the need to worry about escape hatches and make it much simpler for developers to use this construct.
